include/zephyr/rtio/rtio.h

```diff
@@ -750,15 +795,21 @@ static inline void rtio_iodev_cancel_all(struct rtio_iodev *iodev)
  * @param result Integer result code (could be -errno)
  * @param userdata Userdata to pass along to completion
  */
-static inline void rtio_cqe_submit(struct rtio *r, int result, void *userdata)
+static inline void rtio_cqe_submit(struct rtio *r, int result, void *userdata, uint8_t *buffer, uint32_t buffer_len)
```

Comment:
I'd add another helper for the mempool case, rtio_cqe_submit_with_buf(...);

Maybe there's other allocation schemes that come about, so its kind of nice to see the pointer and len, though it does add a 2 word (8-16 bytes) per completion cost.

In io_uring I believe the provided bufs feature instead dumps out a uint16_t index and uint16_t flags pair. The index being the position of the associated block in the buf, this is also doable and might save some precious memory.